waku/waku_rln_relay/rln/rln_interface.nim

```diff
@@ -101,43 +101,43 @@ proc seeded_key_gen*(ctx: ptr RLN, input_buffer: ptr Buffer, output_buffer: ptr
+## output_buffer holds the proof data and should be parsed as [ proof<128> | root<32> | external_nullifier<32> | share_x<32> | share_y<32> | nullifier<32> ]
 ## integers wrapped in <> indicate value sizes in bytes
 ## the return bool value indicates the success or failure of the operation
 
 proc verify*(ctx: ptr RLN,
-                       proof_buffer: ptr Buffer,
-                       proof_is_valid_ptr: ptr bool): bool {.importc: "verify_rln_proof".}
-## proof_buffer has to be serialized as [ proof<128> | root<32> | epoch<32> | share_x<32> | share_y<32> | nullifier<32> | rln_identifier<32> | signal_len<8> | signal<var> ]
+             proof_buffer: ptr Buffer,
+             proof_is_valid_ptr: ptr bool): bool {.importc: "verify_rln_proof".}
+## proof_buffer has to be serialized as [ proof<128> | root<32> | external_nullifier<32> | share_x<32> | share_y<32> | nullifier<32> | signal_len<8> | signal<var> ]
 ## the return bool value indicates the success or failure of the call to the verify function
 ## the verification of the zk proof is available in proof_is_valid_ptr, where a value of true indicates success and false a failure
 
 proc verify_with_roots*(ctx: ptr RLN,
-                      proof_buffer: ptr Buffer,
-                      roots_buffer: ptr Buffer,
-                      proof_is_valid_ptr: ptr bool): bool {.importc: "verify_with_roots".}
-## proof_buffer has to be serialized as [ proof<128> | root<32> | epoch<32> | share_x<32> | share_y<32> | nullifier<32> | rln_identifier<32> | signal_len<8> | signal<var> ]
+                       proof_buffer: ptr Buffer,
+                       roots_buffer: ptr Buffer,
+                       proof_is_valid_ptr: ptr bool): bool {.importc: "verify_with_roots".}
+## proof_buffer has to be serialized as [ proof<128> | root<32> | external_nullifier<32> | share_x<32> | share_y<32> | nullifier<32> | signal_len<8> | signal<var> ]
 ## roots_buffer contains the concatenation of 32 bytes long serializations in little endian of root values
 ## the return bool value indicates the success or failure of the call to the verify function
 ## the verification of the zk proof is available in proof_is_valid_ptr, where a value of true indicates success and false a failure
 
 proc zk_prove*(ctx: ptr RLN,
-            input_buffer: ptr Buffer,
-            output_buffer: ptr Buffer): bool {.importc: "prove".}
+               input_buffer: ptr Buffer,
+               output_buffer: ptr Buffer): bool {.importc: "prove".}
 ## Computes the zkSNARK proof and stores it in output_buffer for input values stored in input_buffer
-## input_buffer is serialized as input_data as [ id_key<32> | path_elements<Vec<32>> | identity_path_index<Vec<1>> | x<32> | epoch<32> | rln_identifier<32> ]
+## input_buffer is serialized as input_data as [ identity_secret<32> | user_message_limit<32> | message_id<32> | path_elements<Vec<32>> | identity_path_index<Vec<1>> | x<32> | external_nullifier<32> ]
```

Comment:
We are changing the param definition but this actually depends on the `zerokit/rln` implementation being used, right?

Reply:
addressed in 561a967e